### PR TITLE
feat: store diffs instead of patches for config changes

### PR DESCRIPTION
**Pull request by adityathebe**
resolves https://github.com/flanksource/config-db/issues/154

### CODE REVIEW

File: db/update.go [hunk 1]
```diff
@@ -209,26 +212,60 @@ func SaveResults(ctx *v1.ScrapeContext, results []v1.ScrapeResult) error {
 	return nil
 }
 
-func generateDiff(a, b models.ConfigItem) (*dutyModels.ConfigChange, error) {
-	patch, err := jsonpatch.CreateMergePatch([]byte(*a.Config), []byte(*b.Config))
+// normalizeJSON returns an idented json string
+func normalizeJSON(jsonStr string) (string, error) {
```
Comment (moshloop) [inline db/update.go:216]: This should also have the keys sorted alphabetically

File: go.mod [hunk 2]
```diff
@@ -271,4 +272,5 @@ require (
 replace (
 	sigs.k8s.io/kustomize/api => sigs.k8s.io/kustomize/api v0.12.1
 	sigs.k8s.io/kustomize/kyaml => sigs.k8s.io/kustomize/kyaml v0.13.9
+	github.com/flanksource/duty => /home/gunners/Projects/flanksource/duty
```
Comment (moshloop) [inline go.mod:275]:
```suggestion
```

File: db/update.go [hunk 3]
```diff
@@ -209,26 +212,60 @@ func SaveResults(ctx *v1.ScrapeContext, results []v1.ScrapeResult) error {
...
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonStrIndented), nil
+}
+
+// generateDiff calculates the diff (git style) and patches between the
+// given 2 config items and returns a ConfigChange object if there are any changes.
+func generateDiff(newConf, prev models.ConfigItem) (*dutyModels.ConfigChange, error) {
```
Comment (moshloop) [inline db/update.go:232]: Can we add a fixture/test for this ?

File: db/update.go (hunk 1 above)
Comment (adityathebe) [reply db/update.go:216]: @moshloop the Go `"encoding/json"` package does that by default.

File: db/update.go (hunk 3 above)
Comment (adityathebe) [reply db/update.go]: Added

File: go.mod (hunk 2 above)
Comment (adityathebe) [reply go.mod]: Waiting for the [related duty PR](https://github.com/flanksource/duty/pull/101) to be merged

File: db/update.go [hunk 4]
```diff
@@ -209,26 +212,71 @@ func SaveResults(ctx *v1.ScrapeContext, results []v1.ScrapeResult) error {
...
+	edits := myers.ComputeEdits("", before, after)
+	if len(edits) == 0 {
+		return "", nil
+	}
+
+	diff := fmt.Sprint(gotextdiff.ToUnified("before", "after", before, edits))
+	return diff, nil
+}
+
+// generateConfigChange calculates the diff (git style) and patches between the
```
Comment (moshloop) [inline db/update.go:255]:
```suggestion
// generateJsonPatch calculates a JSON patch between 2 files
```

Comment (adityathebe) [reply db/update.go:255]: @moshloop I think the existing comment is more descriptive. It's not just calculating json patch.

**Pull request merged**